src/deterministic_randomizer.rs

```diff
@@ -147,6 +151,18 @@ where
     }
 }
 
+#[cfg(feature = "zero")]
+impl<R> Zeroize for DeterministicRandomizer<R>
+where
+    R: CryptoRng + RngCore + SeedableRng,
+    <R as SeedableRng>::Seed: Clone,
+{
+    fn zeroize(&mut self) {
+        let seed = <R as SeedableRng>::Seed::default();
+        *self = DeterministicRandomizer::<R>::new(seed);
+    }
+}
+
```

Comment:
Not a fan of this approach, since this is isn't true zeroization and doesn't provide the same guarantees against compiler optimization.

If you can implement `DefaultIsZeroes` on the `Seed` associated type, you could properly zeroize it. However, there's no guarantee that the PRNG state isn't leaky through the reset.

Overall, I think it's a bad idea to handle bespoke zeroization in this way, as it would provide the user with a sense of security that isn't supported by the implementation. I would remove this entirely.